starlite/signature.py

```diff
@@ -97,25 +92,32 @@ def construct_exception(
         -------
         ValidationException | InternalServerException
         """
-        client_errors = []
-        server_errors = []
+        client_errors: List["ErrorDict"] = []
+        server_errors: List["ErrorDict"] = []
+
         for error in exc.errors():
-            if error["loc"][-1] in cls.factory.dependency_name_set:
+            if cls.is_server_error(error):
                 server_errors.append(error)
             else:
                 client_errors.append(error)
-        method = connection.method if isinstance(connection, Request) else "websocket"
+
+        method, url = cls.get_connection_method_and_url(connection)
 
         if client_errors:
-            return ValidationException(
-                detail=f"Validation failed for {method} {connection.url}",
-                extra=client_errors,
-            )
+            return ValidationException(detail=f"Validation failed for {method} {url}", extra=client_errors)
+
+        return InternalServerException(detail=f"A dependency failed validation for {method} {url}", extra=server_errors)
 
-        return InternalServerException(
-            detail=f"A dependency failed validation for {method} {connection.url}",
-            extra=server_errors,
-        )
+    @classmethod
+    def is_server_error(cls, error: "ErrorDict") -> bool:
+        """Check whether validation error is server error"""
+        return error["loc"][-1] in cls.factory.dependency_name_set
+
+    @staticmethod
+    def get_connection_method_and_url(connection: Union[Request, WebSocket]) -> Tuple[str, URL]:
+        """Extract method and URL from Request or WebSocket"""
+        method = "websocket" if isinstance(connection, WebSocket) else connection.method
```

Comment:
> I don't like this part - somewhat hacky.

Yep. You have a better solution?

> Please instead of websocket use the enum member ScopeType.WebSocket

Done.